Automattic/jetpack
# Backup: Storage UI - i2 - Improving storage warning details

Related to #28085

Similar to what have been implemented in [Calypso for disable access to old backups if storage is full](https://github.com/Automattic/wp-calypso/pull/72145), we are adding a similar UI in backup plugin.

## Proposed changes:
- Add days of backups saved below the storage meter pointing to Activity Log in Jetpack Cloud.
- New detailed messages explained the current storage status in the upgrade prompt section.
- Add `Backups stopped` title when backup is full. Also:
  - Replace `See your backups in the cloud` button with `Upgrade your cloud storage`
  - Replace `Or view your most recent restore point` with `Or view your most recent backup`

### Screenshots:
| Description | Screenshot |
|---|---|
| Normal (< 65% used storage) | <img src="https://user-images.githubusercontent.com/1488641/215018271-dcfc77ae-cc15-486d-8f89-049fafddc602.png" width="500"> |
| Warning (65% - 80% used storage) | <img src="https://user-images.githubusercontent.com/1488641/215018314-87da9661-1b90-41b1-9ce2-c9feeb7a9556.png" width="500"> | 
| Critical (80%-100% used storage) | <img src="https://user-images.githubusercontent.com/1488641/215018330-e2be933c-1610-46a3-9aff-ec29acdcb4bb.png" width="500"> |
| Full (100% used storage) | <img src="https://user-images.githubusercontent.com/1488641/215018347-6df6765e-48fd-4ce5-85e6-484bc6f59127.png" width="500"> |
| Full - Over limit (> 100% used storage) | <img src="https://user-images.githubusercontent.com/1488641/215018357-7f2de12a-9370-4e67-b349-fa72d70e327a.png" width="500"> |
| Backups discarded | <img src="https://user-images.githubusercontent.com/1488641/215018370-d538a8a0-4c3c-4c09-b538-ad08e5ba394d.png" width="500"> |

### Other information:

## Jetpack product discussion


* peaFOp-wy-p2
* p1HpG7-jQY-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:
### Unit testing:
You can run unit testing using the following command:
* JS: `jetpack test packages/backup js`
* PHP: `jetpack test packages/backup php`

### Testing instructions:
* Spin up a Jurassic Ninja site with Jetpack Beta, using Backup branch `add/backup-storage-ui-i2` with a backup plan like Jetpack Backup (1 GB or 10 GB).
* Navigate to Jetpack -> VautlPress Backup
* Once the first backup completes, you should see the main page of the plugin, including the storage meter.
* Given it is a new site, it will be ~0GB used and no upsell prompt should be displayed.

To try different scenarios for the storage meter, you could try to upload big files to your site or dispatch the events using [Redux DevTools (chrome extension)](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en). After installing it, follow these steps:
* Open the extension in Chrome
* In the dropdown in the top right, select `jetpack-backup-plugin`
* In the bottom, click on `>_` option to show the dispatcher.
![image](https://user-images.githubusercontent.com/1488641/213371014-0eaa92d6-d057-41ea-8d1f-2f38a26d026e.png)
* It should look like this:
<img src="https://user-images.githubusercontent.com/1488641/213371077-5db3dc19-35cb-4210-9075-71ea2ef236f0.png" width="250">

* Then in the dispatcher editor, you can set the used storage like this (the size is represented in bytes):
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* So if you have a site with 10 GB, you can use the following sizes to test the different scenarios:
  * Normal (< 65%): 1073741824 (1GB)
  * Warning (65% - 80%): 7516192768 (7GB)
  * Critical (80%-100%): 9663676416 (9GB)
  * Full (> 100%): 10737418240 (10 GB)
 
### Test data
* Normal usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 1073741824,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* Warning usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 7516192768,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* Critical usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 9663676416,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* Full usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 7,
        daysOfBackupsSaved: 7
    }
}
```

* Backups discarded state
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 7,
        daysOfBackupsSaved: 13
    }
}
```
